src/_P050_TCS34725.ino

```diff
@@ -162,17 +176,48 @@ boolean Plugin_050(byte function, struct EventStruct *event, String& string)
         addLog(LOG_LEVEL_DEBUG, F("Found TCS34725 sensor"));
 
         uint16_t r, g, b, c;
+        float value4 = 0.0f;
 
         tcs.getRawData(&r, &g, &b, &c);
-        tcs.calculateColorTemperature(r, g, b);
-        tcs.calculateLux(r, g, b);
-
-        UserVar[event->BaseVarIndex]     = r;
-        UserVar[event->BaseVarIndex + 1] = g;
-        UserVar[event->BaseVarIndex + 2] = b;
-        UserVar[event->BaseVarIndex + 3] = tcs.calculateColorTemperature(r, g, b);
-
-        String log = F("TCS34725: Color Temp (K): ");
+        switch (PCONFIG(3)) {
+          case 0:
+            value4 = tcs.calculateColorTemperature(r, g, b);
+            break;
+          case 1:
+            value4 = tcs.calculateColorTemperature_dn40(r, g, b, c);
+            break;
+          case 2:
+            value4 = tcs.calculateLux(r, g, b);
+            break;
+          case 3:
+            value4 = c;
+            break;
+        }
+
+        if (PCONFIG(2) == 1 && c != 0) { // R/G/B normalized to 0..255 (but avoid divide by 0)
+          UserVar[event->BaseVarIndex]     = (float)r / c * 255.0;
+          UserVar[event->BaseVarIndex + 1] = (float)g / c * 255.0;
+          UserVar[event->BaseVarIndex + 2] = (float)b / c * 255.0;
```

Comment:
Please mark these as `255.0f` otherwise the compiler will add an extra `static_cast<float>(255.0)` equivalent to the code.

Reply:
Ah, will fix, I copied that code almost verbatim from the Adafruit library.